tensorflow/core/kernels/mkl_qmatmul_op.cc

```diff
@@ -0,0 +1,872 @@
+      scaled_bias_ = nullptr;
+    }
+    if (this->comp_bias_ != nullptr) {
+      delete this->comp_bias_;
+      comp_bias_ = nullptr;
+    }
+  }
+
+  float* GetCompBiasBuffer(int size) {
+    if (comp_bias_ == nullptr) {
+      comp_bias_ = new float[size];
+    }
+    return comp_bias_;
+  }
+
+  explicit MklDnnQuantizedMatMulOp(OpKernelConstruction* context)
+      : OpKernel(context) {
+    string mode_string;
+    OP_REQUIRES_OK(context, context->GetAttr("input_quant_mode", &mode_string));
+    if (mode_string == "MIN_FIRST") {
+      mode_ = QUANTIZE_MODE_MIN_FIRST;
+    } else if (mode_string == "SCALED") {
+      mode_ = QUANTIZE_MODE_SCALED;
+    }
+  }
+
+  void Compute(OpKernelContext* context) override {
+    try {
+      // Input tensors
+      const Tensor& src_tensor = MklGetInput(context, kInputIndexSrc);
+      const Tensor& weight_tensor = MklGetInput(context, kInputIndexWeight);
+      const Tensor& bias_tensor = MklGetInput(context, kInputIndexBias);
+
+      MklDnnShape src_mkl_shape, weight_mkl_shape;
+      GetMklShape(context, kInputIndexSrc, &src_mkl_shape);
+      GetMklShape(context, kInputIndexWeight, &weight_mkl_shape);
+      OP_REQUIRES(context, weight_mkl_shape.IsMklTensor() == false,
+                  errors::InvalidArgument("Weight should not be in "
+                                          "MKL Layout"));
+
+      MklDnnData<Tinput> src(&cpu_engine_);
+      MklDnnData<Tweight> weight(&cpu_engine_);
+
+      memory::dims src_dims, weight_dims;
+      memory::dims dst_dims_tf_order, dst_dims_mkl_order;
+
+      // Get shapes of input tensors in MKLDNN order
+      auto src_tf_shape = src_mkl_shape.IsMklTensor()
+                              ? src_mkl_shape.GetTfShape()
+                              : src_tensor.shape();
+      auto weight_tf_shape = weight_mkl_shape.IsMklTensor()
+                                 ? weight_mkl_shape.GetTfShape()
+                                 : weight_tensor.shape();
+
+      src_dims = TFShapeToMklDnnDims(src_tf_shape);
+      weight_dims = TFShapeToMklDnnDims(weight_tf_shape);
+      dst_dims_mkl_order = {static_cast<int>(src_tf_shape.dim_size(0)),
+                            static_cast<int>(weight_tf_shape.dim_size(1))};
+
+      // Weight dims need to be reversed to create inner-product forward
+      // descriptor
+      weight_dims = {static_cast<int>(weight_tf_shape.dim_size(1)),
+                     static_cast<int>(weight_tf_shape.dim_size(0))};
+
+      // Create memory for user data.
+      // Describe how the inputs and outputs of inner-product look like. Also
+      // specify buffers containing actual input and output data.
+      Tensor* dst_tensor = nullptr;
+      auto input_output_fmt = memory::format::nc;
+
+      // If input is in MKL layout, then simply take input layout; otherwise,
+      // construct input TF layout. For TF layout, although input shape
+      // (src_dims) required is in MKLDNN order, the layout is Tensorflow's
```

Comment:
There probably was a misunderstanding. I prefer MKL-DNN over MKLDNN. Could you please change all MKLDNN to MKL-DNN?
```suggestion
      // (src_dims) required is in MKL-DNN order, the layout is Tensorflow's
```